File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/entities/Deck.kt
package com.synergyboat.flashcardAi.domain.entities

import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
data class Deck(
    var id: Long? = null,
    var name: String = "",
    var description: String = "",
    @kotlinx.serialization.Contextual
    var createdAt: Date? = null,
    @kotlinx.serialization.Contextual
    var updatedAt: Date? = null,
    var flashcards: List<Flashcard> = emptyList()
)

File: ios/FlashcardAI/Models/Deck.swift
import Foundation

struct Deck: Codable {
    let id: Int?
    let name: String
    let description: String
    let createdAt: Date
    let updatedAt: Date
    var flashcardCount: Int?
    
    init(id: Int? = nil, name: String, description: String = "", createdAt: Date = Date(), updatedAt: Date = Date(), flashcardCount: Int? = nil) {
        self.id = id
        self.name = name
        self.description = description
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.flashcardCount = flashcardCount
    }
} 